inventory/src/ts/components/item-group.ts

```diff
@@ -0,0 +1,115 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+import {Inventory, Item, itemType} from 'camelot-unchained';
+
+/**
+ * ItemGroup
+ */
+export class ItemGroup {
+  item: Item;
+  itemIDs: string[];
+  quantity: number;
+
+  constructor(item: Item) {
+    this.item = item;
+    this.quantity = 0;
+    this.itemIDs = [];
+    this.addItemID(item.id);
+  }
+
+  getFirstItemID(): string {
+    return this.itemIDs.length > 0 ? this.itemIDs[0] : null;
+  }
+
+  /**
+   * Check if the item group contains an item
+   * @param  {string} id - the id of item to look for
+   * @return {boolean} returns true if the item existing in the item group
+   */
+  hasItemID(id: string): boolean {
+    return this.itemIDs.filter((itemID: string) => {
+      return itemID == id;
+    }).length > 0;
+  }
+
+  /**
+   * Adds an item to the item group
+   * @param {string} id - the item to add to item group
+   */
+  addItemID(id: string): void {
+    if (this.hasItemID(id) == false) {
+      this.itemIDs.push(id);
+      this.quantity += 1;
+    }
+  }
+
+  /**
+   * Removes an item from the item group with the given item id
+   * @param {string} id - the item id to remove
+   */
+  removeItemID(id: string): void {
+    if (this.hasItemID(id)) {
+      let itemIndex: number = null;
+      this.itemIDs.forEach((itemID: string, index: number) => {
+        if (itemID == id) {
+          itemIndex = index;
+        }
+      });
+      if (itemIndex != null) {
+        this.itemIDs.splice(itemIndex, 1);
+        this.quantity -= 1;
+      }
+    }
+  }
+
```

Comment:
Same here, just use ``this.itemIDs.indexOf(id)`` to get the index of the itemId in the list, and also may as well use that to determine if it is in the list.  Which leaves:

    const itemIndex: number = this.itemIDs.indexOf(id);
    if (itemIndex > -1) {
      this.itemIDs.splice(itemIndex, 1);
      this.quantity --;
    }